woocommerce/woocommerce-android
# Validate shipping address

Closes: #7257

### Description
This PR validates the shipping address in the order details screen and tracks the information when the address contains errors. Our goal is to enhance the user experience by adding an address validation mechanism similar to the one the shipping labels plugin uses, but first, we want to understand better the impact these changes could have.

### Testing instructions

TC1
1. Open order list
2. Select an order containing a shipping address with errors
3. Check that the errors are tracked in an event similar to this one:

```
🔵 Tracked: order_address_validation_error, Properties: {"error_message":"The phone number is invalid","validation_scenario":"local","order_id":683,"blog_id":202934350,"is_wpcom_store":true,"is_debug":true}
```

TC2
1. Open order list
2. Select an order without a shipping address
3. Check that no errors are tracked

TC3
1. Deactivate the WooCommerce Shipping & Tax plugin in your store
2. Restart the Woo app on your device
3. Open order list
4. Select an order
5. Check that the remote validation is not sent when the plugin is not activated (No request is sent to `/wc/v1/connect/normalize-address/&_method=post`) 

You can use Flipper to check TC3 and verify a request, as the one below never happens
<img width="1024" alt="Screen Shot 2022-08-28 at 14 07 45" src="https://user-images.githubusercontent.com/18119390/187086235-1fa731ff-a584-4d82-aaee-d7370994a15f.png">
